### ISSUE TITLE
sql: v23.1.8: unexpected error during partial index predicate type resolution: column × does not exist

### ISSUE THREAD

**Issue by cockroach-sentry** [cockroachdb/cockroach#108587]
This issue was auto filed by Sentry. It represents a crash or reported error on a live cluster with telemetry enabled.

Sentry Link: [https://cockroach-labs.sentry.io/issues/4387923238/?referrer=webhooks_plugin](https://cockroach-labs.sentry.io/issues/4387923238/?referrer=webhooks_plugin)

Panic Message:

```
partial_index.go:214: unexpected error during partial index predicate type resolution: column × does not exist
(1) assertion failure
Wraps: (2) attached stack trace
  -- stack trace:
  | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.resolvePartialIndexPredicate.func1
  | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/partial_index.go:214
  | runtime.gopanic
  | 	GOROOT/src/runtime/panic.go:884
  | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.wrapColTupleStarPanic
  | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:1721
  | runtime.gopanic
  | 	GOROOT/src/runtime/panic.go:884
  | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).expandStar
  | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/util.go:123
  | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*scope).VisitPre
  | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:1011
  | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*scope).VisitPre
  | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:1023
  | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.WalkExpr
  | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:831
  | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*scope).walkExprTree
  | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:428
  | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*scope).resolveType
  | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:467
  | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).expandStar
  | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/util.go:65
  | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*scope).VisitPre
  | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:1011
  | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*scope).VisitPre.func1
  | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:1038
  | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*scope).VisitPre
  | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:1039
  | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*scope).VisitPre
  | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:1023
  | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.WalkExpr
  | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:831
  | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*ComparisonExpr).Walk
  | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:191
  | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.WalkExpr
  | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:834
  | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*scope).walkExprTree
  | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:428
  | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*scope).resolveAndRequireType
  | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:486
  | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.resolvePartialIndexPredicate
  | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/partial_index.go:217
  | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildPartialIndexPredicate
  | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/partial_index.go:128
  | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*arbiterPredicateHelper).arbiterFilters
  | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/mutation_builder_arbiter.go:631
  | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*arbiterPredicateHelper).predicateIsImpliedByArbiterPredicate
  | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/mutation_builder_arbiter.go:654
  | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*mutationBuilder).inferArbitersFromConflictOrds
  | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/mutation_builder_arbiter.go:213
  | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*mutationBuilder).findArbiters
  | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/mutation_builder_arbiter.go:111
  | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*mutationBuilder).buildInputForDoNothing
  | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/insert.go:701
  | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildInsert
  | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/insert.go:300
  | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildStmt.func2
  | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:344
  | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).processWiths
  | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/with.go:116
  | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildStmt
  | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:343
  | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildStmtAtRoot
  | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:272
Wraps: (3) secondary error attachment
  | column × does not exist
  | (1) candidate pg code: 42703
  | Wraps: (2) attached stack trace
  |   -- stack trace:
  |   | github.com/cockroachdb/cockroach/pkg/sql/catalog/colinfo.NewUndefinedColumnError
  |   | 	github.com/cockroachdb/cockroach/pkg/sql/catalog/colinfo/column_resolver.go:196
  |   | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*scope).FindSourceProvidingColumn
  |   | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:847
  |   | github.com/cockroachdb/cockroach/pkg/sql/catalog/colinfo.ResolveColumnItem
  |   | 	github.com/cockroachdb/cockroach/pkg/sql/catalog/colinfo/column_item_resolver.go:115
  |   | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*scope).VisitPre
  |   | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:1026
  |   | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*scope).VisitPre
  |   | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:1023
  |   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.WalkExpr
  |   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:831
  |   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*ComparisonExpr).Walk
  |   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:191
  |   | github.com/cockroachdb/cockroach/pkg/sql/sem/tree.WalkExpr
  |   | 	github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:834
  |   | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*scope).walkExprTree
  |   | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:428
  |   | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*scope).resolveAndRequireType
  |   | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:486
  |   | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.resolvePartialIndexPredicate
  |   | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/partial_index.go:217
  |   | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildPartialIndexPredicate
  |   | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/partial_index.go:128
  |   | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*arbiterPredicateHelper).arbiterFilters
  |   | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/mutation_builder_arbiter.go:631
  |   | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*arbiterPredicateHelper).predicateIsImpliedByArbiterPredicate
  |   | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/mutation_builder_arbiter.go:654
  |   | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*mutationBuilder).inferArbitersFromConflictOrds
  |   | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/mutation_builder_arbiter.go:213
  |   | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*mutationBuilder).findArbiters
  |   | 	github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/mutation_builder_arbiter.go:111
  |   | github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder....
```

<details>
<summary>Stacktrace (expand for inline code snippets):</summary>

[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/builder.go#L271-L273](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/builder.go#L271-L273)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/builder.go#L342-L344](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/builder.go#L342-L344)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/with.go#L115-L117](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/with.go#L115-L117)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/builder.go#L343-L345](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/builder.go#L343-L345)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/insert.go#L299-L301](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/insert.go#L299-L301)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/insert.go#L700-L702](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/insert.go#L700-L702)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/mutation_builder_arbiter.go#L110-L112](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/mutation_builder_arbiter.go#L110-L112)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/mutation_builder_arbiter.go#L212-L214](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/mutation_builder_arbiter.go#L212-L214)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/mutation_builder_arbiter.go#L653-L655](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/mutation_builder_arbiter.go#L653-L655)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/mutation_builder_arbiter.go#L630-L632](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/mutation_builder_arbiter.go#L630-L632)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/partial_index.go#L127-L129](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/partial_index.go#L127-L129)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/partial_index.go#L216-L218](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/partial_index.go#L216-L218)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/scope.go#L485-L487](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/scope.go#L485-L487)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/scope.go#L427-L429](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/scope.go#L427-L429)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/sem/tree/walk.go#L833-L835](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/sem/tree/walk.go#L833-L835)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/sem/tree/walk.go#L190-L192](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/sem/tree/walk.go#L190-L192)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/sem/tree/walk.go#L830-L832](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/sem/tree/walk.go#L830-L832)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/scope.go#L1022-L1024](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/scope.go#L1022-L1024)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/scope.go#L1038-L1040](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/scope.go#L1038-L1040)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/scope.go#L1037-L1039](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/scope.go#L1037-L1039)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/scope.go#L1010-L1012](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/scope.go#L1010-L1012)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/util.go#L64-L66](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/util.go#L64-L66)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/scope.go#L466-L468](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/scope.go#L466-L468)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/scope.go#L427-L429](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/scope.go#L427-L429)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/sem/tree/walk.go#L830-L832](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/sem/tree/walk.go#L830-L832)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/scope.go#L1022-L1024](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/scope.go#L1022-L1024)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/scope.go#L1010-L1012](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/scope.go#L1010-L1012)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/util.go#L122-L124](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/util.go#L122-L124)
[GOROOT/src/runtime/panic.go#L883-L885](GOROOT/src/runtime/panic.go#L883-L885)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/scope.go#L1720-L1722](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/scope.go#L1720-L1722)
[GOROOT/src/runtime/panic.go#L883-L885](GOROOT/src/runtime/panic.go#L883-L885)
[https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/partial_index.go#L213-L215](https://github.com/cockroachdb/cockroach/blob/2c1d4afd2c2de091a00a4dba4392ceeaf8504a95/pkg/sql/opt/optbuilder/partial_index.go#L213-L215)
</details>

```
pkg/sql/opt/optbuilder/builder.go in pkg/sql/opt/optbuilder.(*Builder).buildStmtAtRoot at line 272
pkg/sql/opt/optbuilder/builder.go in pkg/sql/opt/optbuilder.(*Builder).buildStmt at line 343
pkg/sql/opt/optbuilder/with.go in pkg/sql/opt/optbuilder.(*Builder).processWiths at line 116
pkg/sql/opt/optbuilder/builder.go in pkg/sql/opt/optbuilder.(*Builder).buildStmt.func2 at line 344
pkg/sql/opt/optbuilder/insert.go in pkg/sql/opt/optbuilder.(*Builder).buildInsert at line 300
pkg/sql/opt/optbuilder/insert.go in pkg/sql/opt/optbuilder.(*mutationBuilder).buildInputForDoNothing at line 701
pkg/sql/opt/optbuilder/mutation_builder_arbiter.go in pkg/sql/opt/optbuilder.(*mutationBuilder).findArbiters at line 111
pkg/sql/opt/optbuilder/mutation_builder_arbiter.go in pkg/sql/opt/optbuilder.(*mutationBuilder).inferArbitersFromConflictOrds at line 213
pkg/sql/opt/optbuilder/mutation_builder_arbiter.go in pkg/sql/opt/optbuilder.(*arbiterPredicateHelper).predicateIsImpliedByArbiterPredicate at line 654
pkg/sql/opt/optbuilder/mutation_builder_arbiter.go in pkg/sql/opt/optbuilder.(*arbiterPredicateHelper).arbiterFilters at line 631
pkg/sql/opt/optbuilder/partial_index.go in pkg/sql/opt/optbuilder.(*Builder).buildPartialIndexPredicate at line 128
pkg/sql/opt/optbuilder/partial_index.go in pkg/sql/opt/optbuilder.resolvePartialIndexPredicate at line 217
pkg/sql/opt/optbuilder/scope.go in pkg/sql/opt/optbuilder.(*scope).resolveAndRequireType at line 486
pkg/sql/opt/optbuilder/scope.go in pkg/sql/opt/optbuilder.(*scope).walkExprTree at line 428
pkg/sql/sem/tree/walk.go in pkg/sql/sem/tree.WalkExpr at line 834
pkg/sql/sem/tree/walk.go in pkg/sql/sem/tree.(*ComparisonExpr).Walk at line 191
pkg/sql/sem/tree/walk.go in pkg/sql/sem/tree.WalkExpr at line 831
pkg/sql/opt/optbuilder/scope.go in pkg/sql/opt/optbuilder.(*scope).VisitPre at line 1023
pkg/sql/opt/optbuilder/scope.go in pkg/sql/opt/optbuilder.(*scope).VisitPre at line 1039
pkg/sql/opt/optbuilder/scope.go in pkg/sql/opt/optbuilder.(*scope).VisitPre.func1 at line 1038
pkg/sql/opt/optbuilder/scope.go in pkg/sql/opt/optbuilder.(*scope).VisitPre at line 1011
pkg/sql/opt/optbuilder/util.go in pkg/sql/opt/optbuilder.(*Builder).expandStar at line 65
pkg/sql/opt/optbuilder/scope.go in pkg/sql/opt/optbuilder.(*scope).resolveType at line 467
pkg/sql/opt/optbuilder/scope.go in pkg/sql/opt/optbuilder.(*scope).walkExprTree at line 428
pkg/sql/sem/tree/walk.go in pkg/sql/sem/tree.WalkExpr at line 831
pkg/sql/opt/optbuilder/scope.go in pkg/sql/opt/optbuilder.(*scope).VisitPre at line 1023
pkg/sql/opt/optbuilder/scope.go in pkg/sql/opt/optbuilder.(*scope).VisitPre at line 1011
pkg/sql/opt/optbuilder/util.go in pkg/sql/opt/optbuilder.(*Builder).expandStar at line 123
GOROOT/src/runtime/panic.go in runtime.gopanic at line 884
pkg/sql/opt/optbuilder/scope.go in pkg/sql/opt/optbuilder.wrapColTupleStarPanic at line 1721
GOROOT/src/runtime/panic.go in runtime.gopanic at line 884
pkg/sql/opt/optbuilder/partial_index.go in pkg/sql/opt/optbuilder.resolvePartialIndexPredicate.func1 at line 214
```

### Tags

| Tag | Value |
| --- | --- |
| Command | start-single-node |
| Environment | v23.1.8 |
| Go Version | go1.19.10 |
| Platform | linux amd64 |
| Distribution | CCL |
| Cockroach Release | v23.1.8 |
| Cockroach SHA | 2c1d4afd2c2de091a00a4dba4392ceeaf8504a95 |
| # of CPUs | 8 |
| # of Goroutines | 400 |

Jira issue: CRDB-30533

**Comment by yuzefovich** [comment]
Seems similar to #96924, but the fix should already be in 23.1.8.

**Comment by yuzefovich** [comment]
Closing as unactionable.
